### PR TITLE
BE works but FE needs some fix

### PR DESCRIPTION
**Pull request by shinheechae**

### CODE REVIEW

**Comment by losrobbos** [comment]
A little too many files to check :D 
Can you remove the "node_modules" from git?
In case the node_modules folder was already commited and pushed, we can remove it like this:
`git rm -r --cached node_modules` => this will delete the folder from .git, but keeps it in your directory!
Afterwards you can commit & push to remove the folder from git

Then I will check the remaining files  ;)

**Comment by shinheechae** [comment]
my apology, I just saw your message and untracked the node_modules. thanks! :)

**Comment by losrobbos** [comment]
> my apology, I just saw your message and untracked the node_modules. thanks! :)

Aaaah nice! So much easier now for me to check :tada: 

And congrats: Being able to revert stuff we did not want to do in the first place, is a common task, so was well worth the training :D

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+node_modules
```
Comment (losrobbos) [inline .gitignore:1]: :partying_face:

File: client/src/components/PrivateRoute.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import React, { useContext } from 'react';
+import loginContext from '../context/loginContext';
+import { Redirect, Route } from 'react-router';
+
+const PrivateRoute = ({component: Component, ...rest}) => {
+    const {user} = useContext(loginContext);
+
+    return (
+      <Route {...rest} render={
```
Comment (losrobbos) [inline client/src/components/PrivateRoute.js:9]: YESSS! Very nicely realized the conditional rendering of the route / redirect. :100:

File: db-connect.js [hunk 3]
```diff
@@ -0,0 +1,7 @@
+import mongoose from 'mongoose'
+
+const MONGO_URI = "mongodb+srv://shinhee:melatonin@shinhee.fglsw.mongodb.net/hasheduser2?retryWrites=true&w=majority"
```
Comment (losrobbos) [inline db-connect.js:3]: This we might outsource to .env, but just splitting hairs at this point :)

File: server.js [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+        next();
+    }
+
+    catch(err) {
+        next(err);
+    }
+}
+
+app.get('/', (req, res)=> {
+    res.send(`<h3>Shinhee's calling</h3>`)
```
Comment (losrobbos) [inline server.js:42]: Heard your call! :)

File: server.js [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+            return next(new Error(`'${username}' does not exist`));
+        };
+
+        const loginSuccess = await bcrypt.compareSync(password, foundUser.password);
+    
+        if (!loginSuccess) {
+            return next(new Error('wrong password'));
+        };
+             
+        let token = await jwt.sign(foundUser.toJSON(), secret);
```
Comment (losrobbos) [inline server.js:71]: Nicely kept the password out of the token! :1st_place_medal:

File: server.js [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+        let sessionInSec = 1000*60*30;
+        res.cookie('token', token, {maxAge: sessionInSec, httpOnly: true});
+        console.log('logged in successfully');
+        res.json( foundUser );
+    
+        } 
+    catch(err) {
+        next(err);
+    }
+    }
```
Comment (losrobbos) [inline server.js:81]: in this file the code formatting is a bit off. Usually it helps to apply code formatting in VsCode, when the code indentation gets a bit messy. But that's a minor ;)

**Comment by losrobbos** [comment]
Nice! I just checked the backend part in more detail. And I was very surprised how you kept the password out of the token. Very elegant solution :+1: 

At this point you are already prepared for the final project to come! Keep rockin....

File: server.js (hunk 5 above)
Comment (shinheechae) [reply server.js:71]: Thank you sooooo much for your detailed review!! wish you a lovely weekend :D

**Pull request closed without merging**